src/Tvheadend.cpp

```diff
@@ -2603,12 +2618,24 @@ bool CTvheadend::ParseEvent ( htsmsg_t *msg, bool bAdd, Event &evt )
     evt.SetSummary(str);
   if ((str = htsmsg_get_str(msg, "description")) != NULL)
     evt.SetDesc(str);
-  if (evt.GetDesc().empty() && m_conn->GetProtocol() >= 32)
-  {
-    // If no description is available, use summary as description. This was done by tvh prior to htsp version 32.
-    evt.SetDesc(evt.GetSummary());
-    // No duplicate description/summary
-    evt.SetSummary("");
+  /* 
+    Due to changes in HTSP v32, try to use the summary
+    as the description. If the summary is empty, try to use
+    the subtitle as the description. Also clear the respective
+    entries to avoid duplicate information being displayed.
+
+    This was done by TVHeadend prior to HTSP v32.
+  */
+  else if (evt.GetDesc().empty() && m_conn->GetProtocol() >= 32)
```

Comment:
Why else ifand not just if? you might have gotten empty desc above...

Reply:
I had it as else if since we only really want the code to happen if the description is empty. Its only going to do one or the other and it'd be less efficient in my opinion to have it run two checks.